component/config.go

```diff
@@ -31,46 +33,131 @@ type ConfigValidator interface {
 // ValidateConfig validates a config, by doing this:
 //   - Call Validate on the config itself if the config implements ConfigValidator.
 func ValidateConfig(cfg Config) error {
-	return validate(reflect.ValueOf(cfg))
+	var err error
+	errs := validate(reflect.ValueOf(cfg))
+
+	for _, suberr := range errs {
+		if suberr.err != nil {
+			if suberr.path != "" {
+				err = errors.Join(err, fmt.Errorf("%s: %w", suberr.path, suberr.err))
+			} else {
+				err = errors.Join(err, suberr.err)
+			}
```

Comment:
nit: We could move this logic inside a `Error` method for `pathError`

Reply:
I like that. Done.